### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by spinetree**

### CODE REVIEW

**Review by SeattleChris — COMMENTED** [review]
I see you started updating your HTML to use more semantic tags. The tags like header, main, footer, aside, section, article, nav, and etc can be very helpful. 

You should update your CSS selectors to target the semantic HTML tags. It is best if you can limit your use of class and id attributes. While there certainly will be times that these are the best choice, this approach should not be overused.

**Comment by spinetree** [comment]
I'm confused by this - if I start targeting, say, section, aren't i creating a substantially less extensible codebase? Meaning anyone using those semantic tags in future work will now have their stuff inherit a bunch of styles they don't want? 

I'm basing most of my thinking about this stuff from https://www.keycdn.com/blog/oocss which points to using a lotta classes, using ids sparingly, and .. I don't know if they'd ever really advocate pointing CSS at basic HTML elements unless they were inside of specific-enough containers to eliminate the chance of mistargeting.

File: index.html [hunk 1]
```diff
@@ -2,23 +2,28 @@
...
-        <section id="a" class="header">A</section>
-        <section id="b" class="sidebar-left-lg">B</section>
-        <section id="c" class="2-col-lg">C</section>
-        <section id="d" class="2-col-lg">D</section>
-        <section id="e" class="2-col 4-col-lg">E</section>
-        <section id="f" class="2-col 4-col-lg">F</section>
-        <section id="g" class="2-col 4-col-lg">G</section>
-        <section id="h" class="2-col 4-col-lg">H</section>
-        <section id="i" class="fullwidth">I</section>
+        <header id="a">A</header>
```
Comment (SeattleChris) [inline index.html:14]: The header should be outside (generally before) the main section.

File: index.html [hunk 2]
```diff
@@ -2,23 +2,28 @@
...
+            <section id="c" class="hero col-2-lg">C</section>
+            <section id="d" class="hero col-2-lg">D</section>
+        </section>
+        <section class="articles">
+            <article id="e" class="col-2 col-4-lg">E</article>
+            <article id="f" class="col-2 col-4-lg">F</article>
+            <article id="g" class="col-2 col-4-lg">G</article>
+            <article id="h" class="col-2 col-4-lg">H</article>
+        </section>
+        <footer id="i" class="">I</footer>
```
Comment (SeattleChris) [inline index.html:26]: The footer should be outside (generally after) the main section.

File: index.html [hunk 3]
```diff
@@ -2,23 +2,28 @@
...
-        <section id="b" class="sidebar-left-lg">B</section>
-        <section id="c" class="2-col-lg">C</section>
-        <section id="d" class="2-col-lg">D</section>
-        <section id="e" class="2-col 4-col-lg">E</section>
-        <section id="f" class="2-col 4-col-lg">F</section>
-        <section id="g" class="2-col 4-col-lg">G</section>
-        <section id="h" class="2-col 4-col-lg">H</section>
-        <section id="i" class="fullwidth">I</section>
+        <header id="a">A</header>
+        <aside id="b" class="">B</aside>
```
Comment (SeattleChris) [inline index.html:15]: Generally, the aside could be inside or outside of the main section. I typically default to it being outside of main unless I have a clear reason why it seems like it is a part of the core/main content of the site.

File: css/layout.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
+body {}
```
Comment (SeattleChris) [inline css/layout.css:1]: You should not have an empty ruleset.

File: css/layout.css [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+footer {
+    clear: both;
+}
+
+@media (min-width:768px) {
+
+    main {
+        max-width: 960px;
+        min-width: 960px;
```
Comment (SeattleChris) [inline css/layout.css:22]: It is often nicer to allow a range between the min-width and max-width to allow the page to adapt the size the user prefers. If your intent is to make the page exactly this size, then it is better to use the width rule. When the max-width and min-width are the same value it creates an opportunity for odd and unexpected consequences.

**Review by SeattleChris — COMMENTED** [review]
See my inline notes for some specific details. Outside of those, there are things in the modules.css file that seem to be more about the page layout. Admittedly, there can be a lot of grey area and personal choice when it comes to using the SMACSS approach. It is admittedly hard to think of what belongs in modules when the goal here was to make a very basic wireframe. It would be more obvious if this project had some JavaScript and if there were smaller elements that might be inside or adjacent to the article elements you have here.

**Comment by SeattleChris** [comment]
> I'm basing most of my thinking about this stuff from https://www.keycdn.com/blog/oocss which points to using a lotta classes, using ids sparingly, and .. I don't know if they'd ever really advocate pointing CSS at basic HTML elements 

I notice that this approach was first proposed in 2008. For context, HTML5 initial release was not until 2014, and that is when the semantic tags I mentioned were added. Prior to that, you would have to use div tags for those and there were a TON of problems with creating clear CSS rules and clear page structure in your HTML. There were a number of different ideas for how to have a clear structure and design approach during this era of web development. 

Prior to HTML5, each of the different approaches had their strengths, but with so many different ideas it was hard to have consistency. To make matters worse, even amongst developers following the same logical process, they might choose similar but slightly different values for class names, like the difference between "nav" and "navigation" or something like that for any class attribute. This inconsistency causes huge problems for accessibility (such as people using software to read the page out loud and using non-mouse ways to decide which part of the page to read or how to click on links). During this era, web developers who worked hard towards more accessible design discovered that it had huge pay-offs in their page structure, ease of updating with adding content or even features not in the original page design, SEO improvements, maintainability & future-proofing, and frankly easier to use for everyone regardless if they had any accessibility concerns. 
 
Looking at the article I feel many of the benefits it lists for the approach is advocating are also achieved using the semantic tags in HTML5. Additionally, using these semantic tags as they are intended makes it much easier for screen reader software and other accessibility concerns, search engine web-crawlers (so better placement on search engines), and clarity when reading and/or updating the code. 

That said, even with good use of the semantic tags added in HTML5, there are options for how to structure your CSS rules and HTML page. If you are making good use of the semantic tags, keep accessibility in your structure/design, and use some of the ideas from that article to expand and fill in the gaps not otherwise covered, it could be useful. There will always be debates about what is best, and skilled reasonable people will still disagree on the details.

**Pull request merged**